sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/vaccination/VaccinationEditActivity.java

```diff
@@ -0,0 +1,181 @@
+        BaseEditActivity.startActivity(context, VaccinationEditActivity.class, buildBundle(rootUuid));
+    }
+
+    public static void startActivity(Context context, String rootUuid, boolean finishInsteadOfUpNav) {
+        BaseEditActivity.startActivity(context, VaccinationEditActivity.class, buildBundle(rootUuid, finishInsteadOfUpNav));
+    }
+
+    @Override
+    protected VaccinationEntity queryRootEntity(String recordUuid) {
+        return DatabaseHelper.getVaccinationDao().queryUuid(recordUuid);
+    }
+
+    @Override
+    protected VaccinationEntity buildRootEntity() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        boolean result = super.onCreateOptionsMenu(menu);
+        getSaveMenu().setTitle(R.string.action_save_vaccination);
+        return result;
+    }
+
+    @Override
+    public List<PageMenuItem> getPageMenuData() {
+        List<PageMenuItem> menuItems = PageMenuItem.fromEnum(VaccinationSection.values(), getContext());
+        return menuItems;
+    }
+
+    @Override
+    protected BaseEditFragment buildEditFragment(PageMenuItem menuItem, VaccinationEntity activityRootData) {
+        VaccinationSection section = VaccinationSection.fromOrdinal(menuItem.getPosition());
+        BaseEditFragment fragment;
+        switch (section) {
+
+            case VACCINATION_INFO:
+                fragment = VaccinationEditFragment.newInstance(activityRootData);
+                break;
+            case HEALTH_CONDITIONS:
+                fragment = VaccinationEditHealthConditionsFragment.newInstance(activityRootData);
+                break;
+            default:
+                throw new IndexOutOfBoundsException(DataHelper.toStringNullable(section));
+        }
+
+        return fragment;
+
+    }
+
+    @Override
+    protected int getActivityTitle() {
+        return R.string.heading_vaccination_edit;
+    }
+
+    @Override
+    public void saveData() {
+        if (saveTask != null) {
+            NotificationHelper.showNotification(this, WARNING, getString(R.string.message_already_saving));
+            return;
+        }
+
+        final VaccinationEntity vaccinationEntity = getStoredRootEntity();
+
+        try {
+            FragmentValidator.validate(getContext(), getActiveFragment().getContentBinding());
+        } catch (ValidationException e) {
+            NotificationHelper.showNotification(this, ERROR, e.getMessage());
+            return;
+        }
+
+        saveTask = new SavingAsyncTask(getRootView(), vaccinationEntity) {
+
+            @Override
+            protected void onPreExecute() {
+                showPreloader();
+            }
+
+            @Override
+            public void doInBackground(TaskResultHolder resultHolder) throws DaoException {
+                final VaccinationEntity savedVaccination = DatabaseHelper.getVaccinationDao().saveAndSnapshot(vaccinationEntity);
```

Comment:
You shouldn't be required to save the vaccination manually here because it should automatically happen when you save the immunization below.